charms/kfp-api/src/charm.py

```diff
@@ -269,12 +233,54 @@ def _generate_config(self, mysql, os, viz):
             "ML_PIPELINE_VISUALIZATIONSERVER_SERVICE_HOST": viz["service-name"],
             "ML_PIPELINE_VISUALIZATIONSERVER_SERVICE_PORT": viz["service-port"],
         }
-        return config, config_json
+        return config_json
 
-    def _check_leader(self):
-        if not self.unit.is_leader():
-            # We can't do anything useful when not the leader, so do nothing.
-            raise CheckFailedError("Waiting for leadership", WaitingStatus)
+    def _check_container_connection(self, container: Container) -> None:
+        """Check if connection can be made with container.
+
+        Args:
+            container: the named container in a unit to check.
+
+        Raises:
+            ErrorWithStatus if the connection cannot be made.
+        """
+        if not container.can_connect():
+            raise ErrorWithStatus("Pod startup is not complete", MaintenanceStatus)
+
+    def _upload_files_to_container(self, config_json):
+        """Upload required files to container."""
+        try:
+            self._check_container_connection(self.container)
+        except ErrorWithStatus as error:
+            self.model.unit.status = error.status
+            raise error
+        try:
+            with open("src/sample_config.json", "r") as sample_config:
+                file_content = sample_config.read()
+                self.container.push(
+                    f"{CONFIG_DIR}/{SAMPLE_CONFIG}",
+                    file_content,
+                    make_dirs=True,
+                )
+        except ErrorWithStatus as error:
+            self.logger.error("Failed to upload sample config to container.")
+            raise error
+        try:
+            file_content = yaml.dump(config_json, allow_unicode=True)
```

Comment:
yaml dump json :D
I see what it does, but counter intuitive

Reply:
also, does it really dump?
you want loads or ?

Reply:
Fixed.